theano/gpuarray/tests/test_linalg.py

```diff
@@ -191,3 +199,97 @@ def test_invalid_input_fail_negative_definite(self):
         A_val = -M_val.dot(M_val.T)
         fn = self.get_gpu_cholesky_func(True, False)
         self.assertRaises(LinAlgError, fn, A_val)
+
+
+class TestMagma(unittest.TestCase):
+
+    def setUp(self):
+        if not config.magma.enabled:
+            self.skipTest('Magma is not enabled, skipping test')
+
+    def test_gpu_matrix_inverse(self):
+        A = theano.tensor.fmatrix("A")
+
+        fn = theano.function([A], gpu_matrix_inverse(A), mode=mode_with_gpu)
+        N = 1000
+        A_val = rand(N, N)
```

Comment:
As your code only accepts float32 (`fmatrix` in Python, `GA_FLOAT` checking in C), always use float32 inputs, so that tests can pass even on 64 bits machines (where floatX is float64 by default). Here:
```python
A_val = rand(N, N).astype('float32')
```